utils/gear-tasks/tests/with_client.rs

```diff
@@ -0,0 +1,193 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use gear_node_testing::client::{Client as TestClient, TestClientBuilder, TestClientBuilderExt};
+use gear_tasks::GearTasksRunner;
+use sc_client_api::UsageProvider;
+use sp_api::HashingFor;
+use sp_externalities::Extensions;
+use sp_state_machine::{Ext, OverlayedChanges};
+use std::{
+    sync::{Arc, OnceLock},
+    time::Duration,
+};
+use vara_runtime::{Block, ProcessingTasksAmount};
+
+static CLIENT: OnceLock<Arc<TestClient>> = OnceLock::new();
+
+struct BackendExternalities {
+    extensions: Extensions,
+    overlay: OverlayedChanges<HashingFor<Block>>,
+}
+
+impl Default for BackendExternalities {
+    fn default() -> Self {
+        let mut overlay = OverlayedChanges::default();
+        // emulate actual runtime behavior
+        overlay.enter_runtime().unwrap();
+
+        Self {
+            extensions: Default::default(),
+            overlay,
+        }
+    }
+}
+
+impl BackendExternalities {
+    fn execute_with<R>(&mut self, f: impl FnOnce() -> R) -> R {
+        let client = CLIENT.get().unwrap();
+        let block_hash = client.usage_info().chain.best_hash;
+        let state = client.state_at(block_hash).unwrap();
+
+        let mut ext = Ext::new(&mut self.overlay, &state, Some(&mut self.extensions));
+        sp_externalities::set_and_run_with_externalities(&mut ext, || {
+            gear_tasks::gear_tasks::reinit(ProcessingTasksAmount::get());
+            f()
+        })
+    }
+}
+
+pub fn init_logger() {
+    let _ = env_logger::Builder::from_default_env()
+        .format(|f, record| {
+            use std::io::Write;
+
+            let current_thread = std::thread::current();
+
+            let level = f.default_styled_level(record.level());
+            let module = record.module_path().unwrap_or_default();
+            let thread_name = current_thread
+                .name()
+                .map(str::to_string)
+                .unwrap_or_else(|| format!("{:?}", current_thread.id()));
+
+            writeln!(f, "[{level:<5} ({thread_name}) {module}] {}", record.args(),)
+        })
+        .try_init();
+}
+
+fn new_test_ext() -> BackendExternalities {
+    CLIENT.get_or_init(|| {
+        let mut client = TestClientBuilder::new().build();
+        // Substrate's `CodeExecutor::call()` has explicit flag to use native execution,
+        // so it's applicable for `NativeElseWasmExecutor`, too.
+        // The flag is always set to `false` in our case, so
+        // we set it to true
+        client.gear_use_native();
```

Comment:
I'm wondering if this is just for testing or even production since the current approach has modified 4 substrate packages for passing this `boolean` ...

```
substrate_test_client::Client 
-> sc_client_api::CallExecutor 
-> sc_client::LocalCallExecutor 
-> sp_state_machine::Client
```

I'd suggest to just introduce a `force_native` field in `sc_executor::NativeElseWasmExecutor` with a public mutable method, then we can do this setting stuff in our own repo

- [gear_node_testing::TestClientBuilderExt::use_gear_native](https://github.com/gear-tech/gear/blob/master/node/testing/src/client.rs#L100) for testing
- [gear_service::new_partial](https://github.com/gear-tech/gear/blob/master/node/service/src/lib.rs#L212) for production

ref https://github.com/gear-tech/polkadot-sdk/pull/4

Reply:
> I'm wondering if this is just for testing

It is testing only

Reply:
> I'd suggest to just introduce a force_native field in sc_executor::NativeElseWasmExecutor with a public mutable method

Done. Check again, please